planning/motion_velocity_planner/autoware_motion_velocity_road_user_stop_module/src/road_user_stop_module.hpp

```diff
@@ -0,0 +1,196 @@
+    const autoware_internal_planning_msgs::msg::PathWithLaneId::ConstSharedPtr msg);
+
+  // helper functions
+  bool isTargetObject(const uint8_t label) const;
+  bool isObjectOnRoad(
+    const PredictedObject & object,
+    const std::vector<autoware_utils_geometry::Polygon2d> & relevant_polygons) const;
+  bool isNearCrosswalk(
+    const geometry_msgs::msg::Point & position, const lanelet::LaneletMapPtr & lanelet_map) const;
+  bool isOppositeTrafficUser(
+    const PredictedObject & object, const lanelet::ConstLanelet & lanelet) const;
+
+  std::pair<
+    std::vector<autoware_utils_geometry::Polygon2d>,
+    std::vector<autoware_utils_geometry::Polygon2d>>
+  getRelevantLanelets(const std::shared_ptr<const PlannerData> planner_data) const;
+
+  std::vector<autoware_utils_geometry::Polygon2d> getTrajectoryPolygons(
+    const std::vector<TrajectoryPoint> & decimated_traj_points,
+    const autoware::vehicle_info_utils::VehicleInfo & vehicle_info,
+    const geometry_msgs::msg::Pose & current_ego_pose, const double lat_margin,
+    const bool enable_to_consider_current_pose, const double time_to_convergence,
+    const double decimate_trajectory_step_length) const;
+
+  std::optional<geometry_msgs::msg::Point> planStop(
+    const std::shared_ptr<const PlannerData> planner_data,
+    const std::vector<TrajectoryPoint> & trajectory_points,
+    const std::vector<StopObstacle> & stop_obstacles, const double dist_to_bumper);
+
+  std::vector<StopObstacle> filterStopObstacles(
+    const std::shared_ptr<const PlannerData> planner_data,
+    const std::vector<TrajectoryPoint> & traj_points,
+    const std::vector<TrajectoryPoint> & decimated_traj_points,
+    const std::vector<autoware_utils_geometry::Polygon2d> & decimated_traj_polygons,
+    const std::vector<autoware_utils_geometry::Polygon2d> & polygons_for_vru,
+    const std::vector<autoware_utils_geometry::Polygon2d> & polygons_for_opposing_traffic,
+    const rclcpp::Time & current_time, const double dist_to_bumper);
+
+  void updateTrackedObjects(
+    const std::vector<std::shared_ptr<PlannerData::Object>> & objects,
+    const rclcpp::Time & current_time);
+
+  bool hasMinimumDetectionDuration(
+    const std::string & object_id, const rclcpp::Time & current_time) const;
+
+  lanelet::ConstLanelets getIntersectionLanelets(
+    const lanelet::ConstLanelet & intersection_lanelet,
+    const lanelet::LaneletMapPtr & lanelet_map) const;
+
+  MarkerArray createDebugMarkerArray() const;
+
+  MarkerArray createLaneletPolygonsMarkerArray(
+    const lanelet::ConstLanelets & lanelets, const std::string & ns,
+    const std::array<double, 3> & color) const;
+
+  mutable DebugData debug_data_;
+
+  rclcpp::Clock::SharedPtr clock_{std::make_shared<rclcpp::Clock>(RCL_ROS_TIME)};
+
+  rclcpp::Subscription<autoware_internal_planning_msgs::msg::PathWithLaneId>::SharedPtr
+    sub_path_with_lane_id_;
+  autoware_internal_planning_msgs::msg::PathWithLaneId::ConstSharedPtr path_with_lane_id_;
```

Comment:
I think these subscribers should exist on the node side rather than in the plugin.
Also, accordingly, variables like these should be included in the `PlannerData` instead.

https://github.com/autowarefoundation/autoware_core/blob/main/planning/motion_velocity_planner/autoware_motion_velocity_planner_common/include/autoware/motion_velocity_planner_common/planner_data.hpp